model/host/db.go

```diff
@@ -427,3 +427,41 @@ func GetHostsByFromIdWithStatus(id, status, user string, limit, sortDir int) ([]
 	}
 	return hosts, nil
 }
+
+// HostStatsByDistroPipeline returns a pipeline that will group all up hosts by distro
+// and return the count of hosts as well as how many are running tasks
+func HostStatsByDistroPipeline() []bson.M {
+	pipeline := make([]bson.M, 0)
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			StatusKey: bson.M{
+				"$in": evergreen.UphostStatus,
+			},
+		},
+	})
+	pipeline = append(pipeline, bson.M{
+		"$group": bson.M{
+			"_id": bson.M{
+				"distro": "$distro._id",
+				"status": "$" + StatusKey,
+			},
+			"count": bson.M{
+				"$sum": 1,
+			},
+			"tasks": bson.M{
+				"$addToSet": "$" + RunningTaskKey,
+			},
+		},
+	})
+	pipeline = append(pipeline, bson.M{
```

Comment:
append is variadic, so you can just do append(pipeline, bson.M{}, bson.M{}, bson.M{}) but really, I'd just do return []bson.M{}, with things in the literal constructor.

Reply:
done